typing/mode.ml

```diff
@@ -111,23 +138,17 @@ module Lattices = struct
 
       let legacy = Global
 
-      let le a b =
-        match a, b with Global, _ | _, Local -> true | Local, Global -> false
-
-      let join a b =
-        match a, b with
-        | Local, _ | _, Local -> Local
-        | Global, Global -> Global
-
-      let meet a b =
-        match a, b with
-        | Global, _ | _, Global -> Global
-        | Local, Local -> Local
-
       let print ppf = function
         | Global -> Format.fprintf ppf "global"
         | Local -> Format.fprintf ppf "local"
-    end)
+
+      let[@inline] index = function Global -> 0 | Local -> 1
+
+      let[@inline] of_index_exn = function
+        | 0 -> Global
+        | 1 -> Local
+        | _ -> invalid_arg "Locality.of_index_exn"
```

Comment:
Does this check for invalidity matter? It means that the conversion isn’t a no-op. Might be interesting to benchmark with and without this line.